tests/test_iam.py

```diff
@@ -1426,3 +1427,35 @@ def test_set_policy_detached(self):
 
         self.assertEqual(len(resources), 1)
         self.assertIn('test-role-us-east-1', resources[0]['RoleName'])
+
+
+class DeleteRoleAction(BaseTest):
+    def test_delete_role(self):
+        factory = self.replay_flight_data("test_delete_role")
+        policy_doc = json.dumps({
+            "Version": "2012-10-17",
+            "Statement": [{
+                "Effect": "Allow",
+                "Principal": {
+                    "Service": "ec2.amazonaws.com"
+                },
+                "Action": "sts:AssumeRole"
+            }]
+        })
+        client = factory().client("iam")
+        client.create_role(
```

Comment:
if your creating the role, should also be able to set this as a functional test, with the @functional decorator

Reply:
Cool